tests/unit/slack_annotations/core_test.py

```diff
@@ -1,3 +1,220 @@
-class TestHelloWorld:
-    def test_it(self):
-        pass
+import json
+from datetime import UTC, datetime, timedelta
+
+import httpx
+import pytest
+from freezegun import freeze_time
+
+from slack_annotations.core import (
+    MAX_TEXT_LENGTH,
+    SEARCH_HOURS,
+    _format_annotation,
+    _format_annotations,
+    _get_quote,
+    _get_search_after,
+    _trim_text,
+    notify,
+)
+
+
+def test_get_search_after_without_cache():
+    default = "2024-12-01T00:00:00+00:00"
+
+    assert _get_search_after("", default) == default
```

Comment:
There's quite a few separate functions to be tested, so it's probably worth group the tests for each function into a class. Just makes it easier to see where the tests for one function end and the next begins, and also opens up the possibility to localize fixtures into the classes that use them:

```python
class TestGetSearchAfter:
    def test_without_cache(self):
        ...

    ...

class TestNotify:
    ...

class TestFoo:
    ...

...
```

Reply:
I'd recommend doing this in a separate PR, but if we changed `cli.py` to pass an opened file object into `notify()` instead of a `cache_path` string, then we could add a unit test for `get_search_after()` that passes in a `StringIO` and properly tests how it reads the `search_after` value from the JSON file contents.